Source/BoxUtils/SetValue.hpp

```diff
@@ -23,14 +23,21 @@ class SetValue
     Interval m_interval;
 
   public:
-    SetValue(double a_value, Interval a_interval = Interval(0, NUM_VARS - 1))
+    SetValue(double a_value, Interval a_interval = Interval())
         : m_value(a_value), m_interval(a_interval)
     {
     }
 
     template <class data_t> void compute(Cell<data_t> current_cell) const
     {
-        for (int i = m_interval.begin(); i <= m_interval.end(); ++i)
+        int start_var = m_interval.begin();
+        int end_var = m_interval.end();
+        if (m_interval.size() == 0)
+        {
+            start_var = 0;
+            end_var = current_cell.get_num_out_vars() - 1;
+        }
+        for (int i = start_var; i <= end_var; ++i)
```

Comment:
This is is a change in SetValue behaves. Previously SetValue(1., Interval(4,5)) in new versions it assings 1. to every component. To be more confusing SetValue(1, Interval(4, 6)) retains old behaviours and still does nothing.

One could argue as it is not a problem because so code in GRCHombo repo uses only SetValue with one argument for which behaviour was not changed. But if someone has a not official code that calls SetValue with two arguments it will be nasty surprise for them.

The suggestion for the fix is in 34185994c68d3bf2d59a87f70a808313b536108c (dislaimer: not tested and would preferable also add tests). It still breaks the code with a two argument call to SetValue but at least during the compilation not by silent change in a  behaviour.

Reply:
I assume you mean `Interval(5,4)` and `Interval(6,4)` respectively? I think the probability of some user relying on Intervals of non-positive length not covering any components is negligible so I think this is probably ok. Your suggestion, as you say, breaks the compilation of existing code so I am disinclined to it. An alternative would be to make the length of the default interval very large and negative?

Reply:
You are right, I meant Intervals with negative values. Thanks for the correction.

Both versions from pull request and mine break code compatibility The difference is that mine does it at compile time, the one from pull request does it silently without any indication to a user that the behaviour has changed. The latter option is usually big no for me.

Using a negative size of Interval as an indication that the method should loop through all variables breaks also, in my opinion, the principle of least astonishment. If you want to have a perfect backward compatibility use instead a large positive value for this purposes. Something like in d7ccf58b5a5a8a594bba1a6d74500c22db95b585. Although I still don't like that it results in a call std::max for every cell.

Reply:
I'm happy with your latest version. I can't think of a way of avoiding conditionals/mins/maxs in this function. Should I merge it into this branch?

Reply:
Yes, please.

Reply:
Done (with a slightly reworded commit message).